File: DeezerExercice/DeezerExercice/Artist.m
//
//  Artist.m
//  DeezerExercice
//  Copyright (c) 2015 Deezer. All rights reserved.
//

#import "Artist.h"

@implementation Artist

- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
    self = [super init];
    if (self) {
        self.identifier = [[dictionary objectForKey:@"id"] integerValue];
        self.name = [dictionary objectForKey:@"name"];
        self.pictureUrl = [dictionary objectForKey:@"picture"];
    }
    return self;
}
@end

File: DeezerExercice/DeezerExercice/Artist.swift
//
//  Artist.swift
//  DeezerExercice
//
//  Created by Maxime Maheo on 02/04/2019.
//  Copyright © 2019 Deezer. All rights reserved.
//
import Foundation

@objcMembers class ArtistList: NSObject, Codable {
    let data: [Artist]
}

@objcMembers class Artist: NSObject, Codable {
    
    // MARK: - Properties
    var identifier: Int
    var name: String
    var pictureUrlBig: String
    var pictureUrlMedium: String
    var fans: Int
    
    enum CodingKeys: String, CodingKey {
        case identifier = "id"
        case name
        case pictureUrlBig = "picture_big"
        case pictureUrlMedium = "picture_medium"
        case fans = "nb_fan"
    }
}
